### PR TITLE
Categories data added to API documentation

### LINKED ISSUE #400: Document Categories API
We should document this endpoint using a combination of:
- [API Blueprint](https://apiblueprint.org/) for writing the documentation of resources, actions, and URIs
- [Dredd](https://dredd.readthedocs.io/en/latest/) for testing the documentation matches the API
- [Apiary](https://apiary.io/) for browsing the docs

Please reference the journey issue #387 for further instructions or commentary on documenting.

### PR DESCRIPTION
**Pull request by npendery**
Closes #400

### CODE REVIEW

File: api-docs/api.apib [hunk 1]
```diff
@@ -33,6 +33,40 @@ Exchange credentials for `access_token`.
 
 # Group Categories
 
+## Categories [/categories]
+
+### Retrieve categories index [GET]
```
Comment (joshsmith) [inline api-docs/api.apib]: I think this should maybe rewritten as:

> List all categories

**Comment by joshsmith** [comment]
I think ordering should be:
- Create
- Retrieve
- Update
- Delete
- List

Taking Stripe's docs as our guide.

What do you think?

**Comment by joshsmith** [comment]
@npendery I also forgot to remove the `api-docs` directory in my PR. We should be using `blueprint/api.apib` isntead.

**Comment by npendery** [comment]
@JoshSmith I feel List would be first since I tend to think of that as the base of the other operations, but not a big deal. And sounds good about the location.

Will add that order and the location to the journey issue info.

**Comment by joshsmith** [comment]
This looks good to me. Do you want to rebase and squash @npendery and then you can merge it in yourself.

**Pull request merged**